api/src/main/java/me/lokka30/treasury/api/economy/transaction/Transaction.java

```diff
@@ -27,12 +31,19 @@
      * @param newBalance        the new balance to set
      * @param transactionAmount the amount which to deposit/withdraw
      * @param transactionType   the transaction type
+     * @param currency          the currency the transaction was made into
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    public Transaction(final double newBalance, final double transactionAmount, @NotNull final TransactionType transactionType) {
+    public Transaction(
+            final double newBalance,
+            final double transactionAmount,
+            @NotNull final TransactionType transactionType,
+            @NotNull final Currency currency
+    ) {
         this.newBalance = AmountUtils.ensureAtLeastZero(newBalance);
```

Comment:
Since there is an option to support negative amounts of money this doesn't need to be at least 0 right?